core/domain/exp_jobs_one_off.py

```diff
@@ -528,6 +528,95 @@ def reduce(key, values):
+        try:
+            exploration.validate()
+        except Exception as e:
+            logging.error(
+                'Exploration %s failed non-strict validation: %s' %
+                (item.id, e))
+            return
+        size_of_math_svg = 0
+        largest_math_expression = ''
+        list_of_latex_values = []
+        for state in exploration.states.values():
+            html_string = ''.join(state.get_all_html_content_strings())
+            (
+                size_of_math_svg_in_state, largest_math_expression_in_state,
+                list_of_latex_values_in_state) = (
+                    html_validation_service.
+                    extract_math_rich_text_related_information_from_html(
+                        html_string))
+            list_of_latex_values.extend(list_of_latex_values_in_state)
+            largest_math_expression = (
+                max(
+                    largest_math_expression,
+                    largest_math_expression_in_state, key=len)
+            )
+            size_of_math_svg = size_of_math_svg + size_of_math_svg_in_state
+
+        if size_of_math_svg > 0:
+            value_dict = {
+                'exploration_id': item.id,
+                'size_of_math_svgs': size_of_math_svg,
+                'largest_math_expression': largest_math_expression,
+                'latex_values': list(set(list_of_latex_values))
+            }
+            yield (
+                'Information about math-expressions in explorations',
+                value_dict)
+
+    @staticmethod
+    def reduce(key, values):
+        final_values = [ast.literal_eval(value) for value in values]
+        number_of_explorations_having_math = 0
+        largest_math_expression = ''
+        estimated_no_of_batches = 1
+        size_of_math_svgs_in_a_batch = 0
+        for value in final_values:
+            exp_models.ExplorationMathRichTextInfoModel(
+                id=value['exploration_id'],
+                latex_values=value['latex_values'],
+                estimated_max_size_of_images_in_bytes=int(
+                    value['size_of_math_svgs'])).put()
+            number_of_explorations_having_math = (
+                number_of_explorations_having_math + 1)
+            largest_math_expression = (
+                max(
+                    value['largest_math_expression'], largest_math_expression,
+                    key=len))
+            size_of_math_svgs_in_a_batch = (
+                size_of_math_svgs_in_a_batch +
+                int(value['size_of_math_svgs']))
+            if (size_of_math_svgs_in_a_batch >
+                    feconf.MAX_SIZE_OF_MATH_SVG_BATCH_BYTES):
+                size_of_math_svgs_in_a_batch = 0
+                estimated_no_of_batches = estimated_no_of_batches + 1
+        final_value_dict = {
+            'estimated_no_of_batches': estimated_no_of_batches,
+            'largest_math_expression': largest_math_expression,
+            'number_of_explorations_having_math': (
```

Comment:
If we're just outputting interesting information, maybe also output the total number of svgs and the total size of svgs (and any other info you can think of that might be useful).

Reply:
done